mmr-backend/data_manager.py

```diff
@@ -1,129 +1,197 @@
-        recommendations_for_book: Iterable[UserRecommendation] = filter(
-            lambda recommendation: recommendation.has_book(isbn), self.fake_recommendations)
-        return list(map(lambda recommendation: recommendation.to_dict(), recommendations_for_book))
+        self._cur.execute("select ur.id, b1.isbn as isbn1, b2.isbn as isbn2, urc.author, urc.comment, urc.score " +
+                          "from user_recommendations ur " +
+                          "left join books b1 on ur.book1_id = b1.id " +
+                          "left join books b2 on ur.book2_id = b2.id " +
+                          "left join user_recommendation_comments urc on ur.id = urc.recommendation_id " +
+                          "where b1.isbn = %s or b2.isbn = %s", (isbn, isbn))
+
+        result = self._cur.fetchall()
+        recommendations: dict[int, UserRecommendation] = {}
+        for tuple in result:
+            if tuple["id"] not in recommendations.keys():
+                recommendations[tuple["id"]] = UserRecommendation(
+                    (tuple["isbn1"], tuple["isbn2"]),
+                    UserRecommendation.UserComment(
+                        tuple["author"], tuple["comment"], tuple["score"])
+                )
+            else:
+                recommendations[tuple["id"]].add_comment(UserRecommendation.UserComment(
+                    tuple["author"], tuple["comment"], tuple["score"]))
+
+        return list(map(lambda recommendation: recommendation.to_dict(), recommendations.values()))
 
     def create_user_recommendation(self, isbn1: str, isbn2: str, user: str, comment: str) -> bool:
-        book1: Optional[Book] = self.get_book(isbn1)
-        book2: Optional[Book] = self.get_book(isbn2)
-        if not (book1 and book2):
-            return False
-
-        recommendations: list[UserRecommendation] = list(filter(lambda recommendation: recommendation.has_book(
-            isbn1) and recommendation.has_book(isbn2), self.fake_recommendations))
-
-        if len(recommendations):
-            recommendations[0].add_comment(
-                UserRecommendation.UserComment(user, comment))
-        else:
-            self.fake_recommendations.append(UserRecommendation(
-                (isbn1, isbn2), UserRecommendation.UserComment(user, comment)))
-        return True
+        self._cur.execute("with rows as " + 
+                          "(insert into user_recommendations (book1_id, book2_id) " + 
+                          "select b1.id as book1_id, b2.id as book2_id from books b1 " + 
+                          "join books b2 on b1.isbn = %s and b2.isbn=%s " + 
+                          "returning id as recommendation_id, %s as author, %s as comment, 0 as score) " + 
+                          "insert into user_recommendation_comments (recommendation_id, author, comment, score) " + 
+                          "select * from rows " + 
+                          "returning id", (isbn1, isbn2, user, comment))
```

Comment:
This query is wrong since it always creates new recommendations between books when the pair of books is already recommended by other user.

In that case it should just create the comment